### PR TITLE
A small benchmark and dockerfile

### PR DESCRIPTION
**Pull request by TheLortex**
## Added a Dockerfile

This dockerfile will be useful when deploying the server

## Benchmarking

The benchmark is simple for now:
- it starts the server in a fresh context
- then send 128 messages with two parameters: the number of rooms and the maximum number of connections

On my laptop it takes from 1 to 3 seconds.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.PHONY: all bench
+
+all:
+	dune build
+
+bench:
+	@dune exec -- ./bench/bench.exe --json --minimal --nb-exec 3
```
Comment (zapashcanon) [inline Makefile:7]: `bench/bench.exe` ? the `./` is not required

Comment (TheLortex) [reply Makefile]: Oops that's wrong indeed. It's just `dune build @bench` for now.

**Comment by clecat** [comment]
Thank you a lot for this benchmark

File: ci-server/client_routes.ml [hunk 2]
```diff
@@ -51,7 +51,7 @@ struct
           let digest = Digestif.BLAKE2B.hmac_string ~key:salt password in
           let pwd = Digestif.BLAKE2B.of_hex pwd in
           if not (Digestif.BLAKE2B.equal digest pwd) then
-            Dream.json ~status:`Unauthorized {|{"errcode": "M_FORBIDDEN"}|}
+            Dream.json ~status:`Unauthorized {|{"errcode": "M_FORBIDDEN", "error": "Wrong password"}|}
```
Comment (clecat) [inline ci-server/client_routes.ml:54]: Should we really give out that information ?
At least we should reply with a "Wrong usename or password"

File: ci-server/client_routes.ml [hunk 3]
```diff
@@ -42,7 +42,7 @@ struct
           Store.find t.store (Store.Key.v ["users"; user_id; "password"]) in
         match pwd with
         | None ->
-          Dream.json ~status:`Unauthorized {|{"errcode": "M_FORBIDDEN"}|}
+          Dream.json ~status:`Unauthorized {|{"errcode": "M_FORBIDDEN", "error": "Missing password in store"}|}
```
Comment (clecat) [inline ci-server/client_routes.ml:45]: This error case means that the user does not exist, as a password will always be here if a user is here.
Therefore we do not want to send back this message, but more something like "Wrong username or password"

**Review by clecat — CHANGES_REQUESTED** [review]

**Comment by clecat** [comment]
Thank you

**Pull request merged**